### ISSUE TITLE
Enhancement: Add support for new skin file format

### ISSUE THREAD

**Issue by ventorvar** [Markemp/Cryengine-Converter#42]
As pointed out in #36, Star Citizen as of 3.11.x is using a new file format for `.skin` files. This issue is to track progress in working out this new format and ultimately implementing it in `cryengine-converter`.

**Comment by ventorvar** [comment]
The `.chr` file is now also using the `#ivo` format

**Comment by ventorvar** [comment]
Figured out a bit of the header info. Chunks have their own header definitions now for different chunk types it seems. There are also three discrete loaders for `#ivo` files, and they have conflicting "type" checks for similar looking things. 

(I'm using python to work through this, so defs are copied from that)

```python
FILE_SIGNATURE = b'#ivo'

class IvoVersion(IntEnum):
    SC_3_11 = 0x900

class IvoHeader(ctypes.LittleEndianStructure, FileHeaderStructure):
    _fields_ = [
        ("signature", ctypes.c_uint32),        # FILE_SIGNATURE
        ("version", ctypes.c_uint32),           # IvoVersion
        ("num_chunks", ctypes.c_uint32),
        ("chunk_hdr_table_offset", ctypes.c_uint32)
    ]

class DBAChunkHeaderTypes(IntEnum):
    # From  IVO_Loader? .dba files
    Skeleton        = 0x0000300d
    DBAData         = 0x194fbc50
    DBA             = 0xf7351608  # is checked against -0x8cae9f8 in code


class AIMChunkHeaderTypes(IntEnum):
    # From  IVO_Loader2? handles AIM files, .caf?
    Skeleton        = 0x1bbc4103
    BShapes         = 0xf5c6eb5b


class CharacterChunkHeaderTypes(IntEnum):
    """ Types for .chr/.skin """
    # From  IVO_Loader3 seems to handle .chr, .skin
    Physics         = 0x90c687dc
    BShapesGPU      = 0x57a3befd
    MaterialName    = 0x8335674e
    BShapes         = 0x875ccb28
    SkinInfo        = 0x9293b9d8
    SkinMesh        = 0xb875b2d9
    Skeleton        = 0xc201973c
```

**Comment by ventorvar** [comment]
New chunk headers in the ivo file header are simply:

```Python
class IvoCharacterChunkHeader(ctypes.LittleEndianStructure, StructureWithEnums):
    _fields_ = [
        ('type', ctypes.c_uint32),
        ('version', ctypes.c_uint32),
        ('offset', ctypes.c_uint64),
    ]
```

**Comment by Markemp** [comment]
This is gold.  Thank you!  I'm finishing up another project but will try to get a big chunk of this done next weekend

**Comment by Markemp** [comment]
Armature for the Behr rifle.  Seems to match the 3.4 version too!  This is from the .chr file, which has no geometry.  Next will be working on the .skin/.skinm version of this.

![image](https://user-images.githubusercontent.com/12057128/103499921-045eee80-4e07-11eb-899c-f9cc4b22d96d.png)

Overall it's pretty amazing what CIG has done.  The new files are a fraction of the size of the old ones.  This armature went from about 17KB to 4KB.  Cryengine files are pretty verbose, and it looks like they are really tightening up the data structure.  For example, a compiled bone went from 584 bytes to just 68 (not including the bone name).  Amazing.

Merging this into master.  Not quite ready for a new release (also want to clean up some Archeage armature issues), but next weekend.  Hope to get the .skin stuff done by then too.

**Comment by Markemp** [comment]
Closing as this is essentially resolved in 1.3.0.  Will work on Ivo morph shapes in a later version.
